### PR TITLE
Add ADR for storing repo data as JSON blob

### PR DESCRIPTION
**Pull request by richpjames**

### CODE REVIEW

**Review by adamhughes86 — APPROVED** [review]
These both are sensible suggestions

File: doc/architecture/decisions/0003-store-data-in-json-blob.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+We will store the data the application needs in JSON.
+
+## Consequences
+
+- The data can be persisted between deployments. This wouldn't be possible if stored in-memory.
+- We don't have to describe a rigid database structure.
+- We don't have to add any dependencies.
+- Initial deployments will be more simple
+- We won't be able to query the data using SQL
+- It may be harder to keep the data up to date
```
Comment (shuldt) [inline doc/architecture/decisions/0003-store-data-in-json-blob.md:32]: Do we want to add a brief note as to why here

**Review by shuldt — APPROVED** [review]
Looks good, sensible decisions well expressed

**Review by danlivings-dxw — APPROVED** [review]
It might be worth mentioning that we'll need to configure a container volume to enable data persistence across deployments, but otherwise this makes sense to me.

**Comment by richpjames** [comment]
@danlivings-dxw I was thinking initially we could keep the changes committed alongside the code in order for us to avoid containerisation

**Comment by danlivings-dxw** [comment]
@richpjames I'm not sure we'd want to commit what is essentially a file-based database to this repository, as that would leak information about private repositories and would introduce a manual step to making sure that Towtruck is displaying up-to-date information.

I'd imagine an okay way to do this would be a folder named something like `.towtruck` that lives inside the application folder. That way we can add it to `.gitignore` so we don't accidentally commit its contents, and if we do define a `Dockerfile` at some point it will be close enough that it won't need a separate volume.

**Comment by richpjames** [comment]
@danlivings-dxw yeah that's a good point. I wasn't aware of the number of private repositories dxw has. In #27 I've added it as a seed step that can be run on demand.

**Pull request merged**